Convolution module

mabevillar opened the pull request:
- Creation of V1.0 of the convolution module
- Addition of 3 functions in utils.py which are required for running the convolution

Commit:
comments on functions

Commit:
description of functions merged in scripts

Commit:
Upload of convolution module V1.0

Commit:
Convolution V1.0 and NLTHA changes

Commit:
Update in utils.py

Commit:
Update in utils.py

raoanirudh commented:
A couple of suggestions: 
1. For now, we're using the repository [rmtk_data](https://github.com/GEMScienceTools/rmtk_data) to store all the input and output files produced by the rmtk scripts and notebooks. Could you transfer all of the .csv and .gmr files to that repository and make sure that the .out files produced by OpenSees are also written to that repository?
2. The file [~$Hazard.xlsx](https://github.com/GEMScienceTools/rmtk/pull/38/files#diff-3c6741eb974337cebfb4da0b299814f4) is a temporary file; it should be deleted.

Commit:
Merge remote-tracking branch 'gem/master'

Commit:
Moved input files for convolution_module to rmtk_data

Commit:
Add functions for the convolution_module in utils.py file
Included read_hazard, plot_hazard_curve, and read_frag_model
functions to utils.py

raoanirudh commented on rmtk/vulnerability/common/utils.py:
```diff
@@ -1723,3 +1723,104 @@ def convert_array_to_string(array):
+    #Store all the data in the dictionary
+    fragility_model =  {'Damage_states':None,'log_mean':None,'log_stdev':None}
+    fragility_model['Damage_states'] = damage_states
+    fragility_model['log_mean'] = log_mean
```
Please check if this is consistent with the other definitions, which use `fragility_model['logmean']`

raoanirudh commented on rmtk/vulnerability/common/utils.py:
```diff
@@ -1723,3 +1723,104 @@ def convert_array_to_string(array):
+             if isNumber(value):    
+                IMLs.append(float(value))
+      if line[0] == 'Sa':
+         IM_type.append(line[0])
```
Please check the indentation for all lines in this function.

raoanirudh commented on rmtk/vulnerability/common/utils.py:
```diff
@@ -1723,3 +1723,104 @@ def convert_array_to_string(array):
+def plot_hazard_curve(hazard_curve):
+#This function plots the hazard curve
+
+   IMLs = hazard_curve['IMLs']
```
Please check the indentation for all lines in this function.

raoanirudh commented on rmtk/vulnerability/common/utils.py:
```diff
@@ -1723,3 +1723,104 @@ def convert_array_to_string(array):
+def read_frag_model(input_file):    
+#This function reads a fragility model and stores it in a dictionary
+
+    damage_states = []
```
Please see above comment about indentation.

raoanirudh commented:
@mabevillar Is the file [Frag_model.png](https://github.com/mabevillar/rmtk/blob/master/rmtk/vulnerability/derivation_fragility/hybrid_methods/CSM/OutFiles/Frag_model.png) required for the CSM module, or is it just an output file that can be removed?

raoanirudh commented on rmtk/vulnerability/Convolution_module/Convolution.py:
```diff
@@ -0,0 +1,173 @@
+#This function calculates the annual probability of collapse given a hazard 
+#curve and a fragility model
+
+   rate_exceed = []
```
Please check the indentation for all functions in this module ([PEP 8 link for reference](https://www.python.org/dev/peps/pep-0008/#indentation)).

Commit:
Corrected indentation issues in convolution.py and utils.py
Corrected name definition in read_frag_model function (changed
"fragility_model['log_mean']" for "fragility_model['logmean']"

mabevillar commented:
@raoanirudh : The Frag_model.png in the CSM is just an output file. You can delete it.

Commit:
Correction of additional space in line 145 of convolution.py
' median - dispersion' instead of ' median -  dispersion'
 with '#' will be ignored, and an empty message aborts the commit.

raoanirudh commented:
Looks good.
